packages/eslint-plugin-next/lib/rules/no-head-import-in-document.js

```diff
@@ -15,14 +15,16 @@ module.exports = {
         }
 
         const document = context.getFilename().split('pages')[1]
-        if (!document || !path.parse(document).name.startsWith('_document')) {
-          return
+        const { name, dir } = path.parse(document)
```

Comment:
It looks like the `!document` check is still needed here or else `path.parse` can throw this error: 

```
ESLint: 7.24.0

TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at validateString (internal/validators.js:124:11)
    at Object.parse (path.js:1283:5)
    at ImportDeclaration (/Users/jj/dev/vercel/next.js/packages/eslint-plugin-next/lib/rules/no-head-import-in-document.js:18:36)
    at /Users/jj/dev/vercel/next.js/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/Users/jj/dev/vercel/next.js/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/Users/jj/dev/vercel/next.js/node_modules/eslint/lib/linter/node-event-generator.js:256:26)
    at NodeEventGenerator.applySelectors (/Users/jj/dev/vercel/next.js/node_modules/eslint/lib/linter/node-event-generator.js:285:22)
    at NodeEventGenerator.enterNode (/Users/jj/dev/vercel/next.js/node_modules/eslint/lib/linter/node-event-generator.js:299:14)
    at CodePathAnalyzer.enterNode (/Users/jj/dev/vercel/next.js/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:711:23)
```

Reply:
Ah, fixed!